foxpuppet/windows/browser/panel_ui/panel_ui.py

```diff
@@ -0,0 +1,187 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Contains classes for handling Firefox Panel UI (Hamburger menu)."""
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.windows.browser.navbar import NavBar
+from selenium.webdriver.remote.webelement import WebElement
+from typing import Type, Any, TYPE_CHECKING, Optional
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class PanelUI(NavBar):
+    """Handles interaction with Panel UI."""
+
+    if TYPE_CHECKING:
+        from foxpuppet.windows import BrowserWindow
+
+    @staticmethod
+    def create(
+        window: Optional["BrowserWindow"], root: WebElement
+    ) -> Type["PanelUI"] | Any:
+        """Create a Panel UI object.
+
+        Args:
+            window (:py:class:`BrowserWindow`): Window object this region
+                appears in.
+            root
+                (:py:class:`~selenium.webdriver.remote.webelement.WebElement`):
+                WebDriver element object that serves as the root for the
+                Panel UI.
+
+        Returns:
+            :py:class:`PanelUI`: Firefox Panel UI.
+
+        """
+        panel_items: dict = {}
+        _id: str | bool | WebElement | dict = root.get_property("id")
+
+        panel_items.update(PANEL_ITEMS)
+        return panel_items.get(_id, PanelUI)(window, root)
+
+    @property
+    def is_update_available(self) -> bool:
+        """
+        Checks if the Panel UI button indicates a pending Firefox update.
+
+        Returns:
+            bool: True if an update notification (barge) is present, False otherwise.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            barged_status = self.selenium.find_element(
+                *PanelUILocators.PANEL_UI_BUTTON
+            ).get_attribute("barged")
+            return barged_status == "true"
```

Comment:
I know this is called `barged` but lets call it `update_status`. That will make more sense in the future.